addons/account_payment/controllers/portal.py

```diff
@@ -21,6 +21,14 @@ def _invoice_get_page_view_values(self, invoice, access_token, **kwargs):
             partner_id,
             currency_id=invoice.currency_id.id,
         )  # In sudo mode to read the fields of acquirers and partner (if not logged in)
+
+        # Partner's company must match the invoice company.
+        partner_sudo = request.env['res.partner'].sudo().browse(partner_id).exists()
```

Comment:
There is no need to check if the record exists because we just got the partner id from a partner record. However, it'd be better to change line 18 to create a variable with the partner rather than directly its id. That'd avoid browsing the record again.